### PR TITLE
[PART-II] #1018 Apply AppGrid for Tablet/Mobile web

### PR DESCRIPTION
**Pull request by hoangdat**
- [x] Tablet/Mobile web

https://user-images.githubusercontent.com/6462404/196600345-bb5ab43d-2d24-4bbb-860a-3ccba92e038c.mov

### CODE REVIEW

**Comment by dab246** [comment]
Please rebase code

**Review by ManhNTX — APPROVED** [review]

File: lib/features/mailbox/presentation/mailbox_view_web.dart [hunk 1]
```diff
@@ -159,6 +161,12 @@ class MailboxView extends GetWidget<MailboxController> with AppLoaderMixin, Popu
             return _buildUserInformation(context);
           }),
           _buildLoadingView(),
+          AppConfig.appGridDashboardAvailable && _responsiveUtils.isWebNotDesktop(context)
+            ? _buildAppGridDashboard(context)
+            : const SizedBox.shrink(),
+          const SizedBox(height: 8),
+          const Divider(color: AppColor.colorDividerMailbox, height: 0.5, thickness: 0.2),
```
Comment (dab246) [inline lib/features/mailbox/presentation/mailbox_view_web.dart:168]: You should hide it for it if `AppGridDashboard` does not support.

File: docs/configuration/app_grid_configuration.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+- `appName`: The name will be showed in App Grid
+- `icon`: Name of icon was added in `configurations\icons` folder
+- `appLink`: Service URL
+
+2. Enable it in [env.file](https://github.com/linagora/tmail-flutter/blob/master/env.file)
+```
+APP_GRID_AVAILABLE=supported
+```
+If you want to disable it, please change the value to `unsupported`
```
Comment (dab246) [inline docs/configuration/app_grid_configuration.md:48]: Please add a description.
```
Or you can remove it from file `env`
```

File: lib/features/mailbox/presentation/mailbox_view_web.dart [hunk 3]
```diff
@@ -467,4 +475,90 @@ class MailboxView extends GetWidget<MailboxController> with AppLoaderMixin, Popu
...
+        ]
+      )
+    );
+  }
+
+  Widget _buildAppGridInMailboxView(BuildContext context) {
+    return Obx(() {
+      final linagoraApps = controller.mailboxDashBoardController.appGridDashboardController.linagoraApplications.value;
+      if (linagoraApps != null && linagoraApps.apps.isNotEmpty) {
+        return ListView.separated(
```
Comment (dab246) [inline lib/features/mailbox/presentation/mailbox_view_web.dart:550]: If you don't have lines between items, you shouldn't use `ListView.separated`. Please use `ListView.builder`

File: lib/features/mailbox_dashboard/presentation/widgets/app_dashboard/app_list_dashboard_item.dart [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return StatefulBuilder(
+      builder: (BuildContext context, StateSetter setState) {
+        return InkWell(
+          onTap: () => _openApp(widget.app.appUri),
+          onHover: (value) => setState(() => _isHoverItem = value),
+          child: Container(
+            decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(10),
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/widgets/app_dashboard/app_list_dashboard_item.dart:31]: Should use `BorderRadius.all(BorderRadius.circular(10))`

File: lib/features/mailbox/presentation/mailbox_view_web.dart (hunk 1 above)
Comment (hoangdat) [reply lib/features/mailbox/presentation/mailbox_view_web.dart:168]: After fixed: 

image

File: lib/features/mailbox/presentation/model/mailbox_categories.dart [hunk 5]
```diff
@@ -17,6 +18,8 @@ extension MailboxCategoriessExtension on MailboxCategories {
         return 'exchange';
       case MailboxCategories.folders:
         return 'folders';
+      case MailboxCategories.appGrid:
+        return 'appGrid';
```
Comment (ManhNTX) [inline lib/features/mailbox/presentation/model/mailbox_categories.dart:22]: using enum class

Comment (hoangdat) [reply lib/features/mailbox/presentation/model/mailbox_categories.dart:22]: It still is extension before enum class support, please make a move

**Review by ManhNTX — APPROVED** [review]
LGTM

**Review by dab246 — APPROVED** [review]

**Pull request merged**